### DISCUSSION TITLE
Rest API

### DISCUSSION THREAD

**Discussion by irux** [docker-mailserver/docker-mailserver#3354] [Q&A]
Any way to create emails over a rest api? Thanks!

**Comment by polarathene** [comment]
This is not something DMS does, SendGrid can.

**Reply by irux** [reply]
@polarathene Maybe I was not clear enough. I apologize! I want to create email accounts  (mailboxes) over rest api not sending them. So I would like to know if the setup script has some kind of interface over rest api or maybe any guide to connect with https://doc.dovecot.org/admin_manual/doveadm_http_api/

thanks!

**Reply by polarathene** [reply]
We have not implemented any REST API. There was plans to as another maintainer chimed in and linked, but that stalled.

I wanted to eventually port the account management support in DMS to a Rust service that would support both CLI and REST, but I've not been able to find the time.

If you'd like to implement your own REST API, you could probably extend the DMS image (_or volume mount a binary and run it with `user-patches.sh`_) to run a daemon/server that calls the current `setup email ...`  CLI.

You can also find a discussion in an old DMS issue where some ideas where discussed on how to approach it. I recall wanting to provide an API service that could be enabled via an ENV and deferring security/UI to a separate project which would be less maintenance for DMS and flexible for community to maintain with whatever tech stack suited them. So effectively two separate API services (internal and public).

**Comment by casperklein** [comment]
At the moment, there is no API of any kind. However, there is https://github.com/docker-mailserver/docker-mailserver-admin. But there wasn't any progress lately afaik.
